packages/inventory-compliance/src/SystemRulesTable/SystemRulesTable.js

```diff
@@ -276,11 +279,14 @@ class SystemRulesTable extends React.Component {
         let selectedToRemediate;
 
         if (this.selectAllCheckbox(key)) {
-            selectedToRemediate = this.getRules().map((rule) => rule.rowKey);
+            let ruleRowKeys = this.getRules().map((rule) => rule.rowKey);
+            selectedToRemediate = selected ?
+                [ ...new Set([ ...currentlySelectToRemediate, ...ruleRowKeys ]) ] :
+                currentlySelectToRemediate.filter((rowKey) => !ruleRowKeys.includes(rowKey));
```

Comment:
Pretty much exactly what's in `onSelect` but selecting on a different key (`rowKey` instead of `refId`). If this code was expected to be long lived, we should refactor, but it's on the chopping block once Sebastian's rule table changes are finished.